qiskit_experiments/framework/base_experiment.py

```diff
@@ -452,15 +483,16 @@ def _add_job_metadata(self, metadata: Dict[str, Any], jobs: BaseJob, **run_optio
             jobs: the job objects.
             run_options: backend run options for the job.
         """
-        metadata["job_metadata"] = [
-            {
-                "job_ids": [job.job_id() for job in jobs],
-                "experiment_options": copy.copy(self.experiment_options.__dict__),
-                "transpile_options": copy.copy(self.transpile_options.__dict__),
-                "analysis_options": copy.copy(self.analysis_options.__dict__),
-                "run_options": copy.copy(run_options),
-            }
-        ]
+        values = {
+            "job_ids": [job.job_id() for job in jobs],
+            "experiment_options": copy.copy(self.experiment_options.__dict__),
+            "transpile_options": copy.copy(self.transpile_options.__dict__),
+            "run_options": copy.copy(run_options),
+        }
+        if self._analysis is not None:
+            values["analysis_options"] = copy.copy(self.analysis.options.__dict__)
```

Comment:
I don't think this is needed but just left it here to be consistent with previous behavior wrt result DB (otherwise custom analysis options are not saved anywhere there).